src/core/utils/srp_client_buffers.hpp

```diff
@@ -86,6 +86,12 @@ class SrpClientBuffers : public InstanceLocator, private NonCopyable
          */
         kServiceNameSize = OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_SERVICE_NAME_SIZE,
 
+        /**
+         * Array length for service subtype label.
+         *
+         */
+        kServiceSubTypeArrayLength = OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_SERVICE_SUB_TYPE_ARRAY_LENGTH,
```

Comment:
With `OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_SERVICE_SUB_TYPE_ARRAY_LENGTH` being 7, we can register at most 6 subtypes. 

Why not change it as below so that users know exactly how many subtypes they can use?
```
#define OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_SERVICE_MAX_SUB_TYPES 6
```

Reply:
Thanks. Good suggestion. Changed it in new push.